packages/extensions/openapi-to-typespec/src/resource/utils.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  ManagementGroupPath,
+  ManagementGroupScopePrefix,
+  ProvidersSegment,
+  ResourceGroupPath,
+  ResourceGroupScopePrefix,
+  SubscriptionPath,
+  SubscriptionScopePrefix,
+  TenantPath,
+  TenantScopePrefix,
+} from "./constants";
+
+export function getResourceType(path: string): string {
+  const index = path.lastIndexOf(ProvidersSegment);
+  if (index < 0 || path.substring(index + ProvidersSegment.length).includes("/") === false) {
+    const pathToLower = path.toLowerCase();
+    if (pathToLower.startsWith(ResourceGroupScopePrefix.toLowerCase())) return "Microsoft.Resources/resourceGroups";
+    if (pathToLower.startsWith(SubscriptionScopePrefix.toLowerCase())) return "Microsoft.Resources/subscriptions";
+    if (pathToLower.startsWith(TenantScopePrefix.toLowerCase())) return "Microsoft.Resources/tenants";
+    throw `Path ${path} doesn't have resouce type`;
+  }
+
+  return path
+    .substring(index + ProvidersSegment.length)
+    .split("/")
+    .reduce((result, current, index) => {
+      if (index === 0 || index % 2 === 1) return result === "" ? current : `${result}/${current}`;
+      else return result;
+    }, "");
+}
+
+export function getResourceKey(path: string): string {
+  const segments = path.split("/");
+  return segments[segments.length - 1].replace(/^\{(.+)\}$/, "$1");
+}
+
+export function getResourceKeySegment(path: string): string {
+  const segments = path.split("/");
+  return segments[segments.length - 2];
+}
+
+export function getScopePath(path: string): string {
+  const pathToLower = path.toLowerCase();
+
+  const index = pathToLower.lastIndexOf(ProvidersSegment);
+  if (index === 0 && pathToLower.startsWith(ManagementGroupScopePrefix.toLowerCase())) return ManagementGroupPath;
```

Comment:
can be simplified to `pathToLower.startsWith(ManagementGroupScopePrefix.toLowerCase())`
And also, consider to put the exact matches in front and take care of `providers` in the end.
Shall we throw otherwise?